fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/entrylists/FabricRegistryEntryList.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.registry.entrylists;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.WeakHashMap;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.MustBeInvokedByOverriders;
+
+import net.minecraft.registry.entry.RegistryEntryList;
+
+// Injected to RegistryEntryList<T>
+@ApiStatus.NonExtendable
+public interface FabricRegistryEntryList<T> {
+	// creating a cycle in the graph will lead to stack overflow, do with this knowledge what you will
+	Map<RegistryEntryList<?>, Set<RegistryEntryList<?>>> DEPENDENCIES = new WeakHashMap<>();
```

Comment:
I think it should not be public, i.e. it should be moved to a separate impl class